src/backend/parser/parse_type.c

```diff
@@ -361,6 +362,13 @@ typenameTypeMod(ParseState *pstate, const TypeName *typeName, Type typ)
 
 	typmodin = ((Form_pg_type) GETSTRUCT(typ))->typmodin;
 
+	/*
+	 * Find the OID of domain's typmodin function, which is same as its basetype's typmodin OID,
+	 * for domains like smallmoney/money and their UDTs during restore.
```

Comment:
Lets avoid any extension specific comments like "for domains like smallmoney/money and their UDTs during restore"